calendar-bundle/contao/classes/Calendar.php

```diff
@@ -231,6 +231,7 @@ protected function generateFiles($arrFeed)
 					}
 
 					// Override the global page object (#2946)
+					// FIXME: global object is overwritten
```

Comment:
We need to discuss what to do with the places where the global page object is overwritten.

Reply:
I think we need to keep it for backwards compatibility, but we should also make sure that the request we push to the stack has the pageModel attribute set correctly.

Reply:
My previous comment was lost. We should make sure to implement a Feed Page Controller for Calendar Feeds in Contao 5.5. Then this will be solved.

Reply:
I know. Regardless of the switch to a Calendar Feed Page Controller: we likely will need a "Page" stack in the page finder to fix this and the other places.

Reply:
> we likely will need a "Page" stack

Can’t we just use the request_stack? If `$objPage` gets overwritten we have to push a request to the stack anyway I think, and that request should have the correct page model, and the page finder should use the request stack already.

Reply:
Indeed 👍

Reply:
But that won't solve it for the occurrence in `PageError*` for `FrontendIndex`.

Reply:
why not?

Reply:
How do you want to implement it there? I don't think we should clone the request and push it to the stack.